sdk/analytics/consumer-rules.pro

```diff
@@ -0,0 +1 @@
+-keep class com.klaviyo.core.** { *; }
```

Comment:
should this be:
```
-keep class com.klaviyo.analytics.** { *; }
```

Reply:
I was under the impression that you need to declare the modules you _depend_ on, not the module that you are writing the proguard rule in. So if we wanted to protect against obfuscating the core library when a customer is using the analytics module, we need to basically keep any of our dependencies. I could be wrong about that though.
https://medium.com/android-news/proguard-r8-in-the-world-of-modularity-f599650b4553

Reply:
I was going to say the same thing as Kenny. But then I reread that article. And I think Dan has it right -- and this finally clarifies for me why this only happened to apps dependent upon the react native SDK, and I couldn't reproduce it in a straight Android app that depended on the android SDK modules directly.

In the case of react native, using those event constants represents a transitive dependency, and since we don't explicitly reference the event constants in the react native SDK code, the shrinker thinks it can remove them. We reference those constants dynamically via reflection so the shrinker doesn't see it. So they are missing at run time.

So I think it is by virtue of the fact that your proguard consumer rule in the push-fcm module instruct the consumer to keep all analytics classes that the react native test app winds up knowing to keep the event constants when shrunk. 

In that sense, for robustness these proguard rules technically belong in the react native SDK repo too, though it'd be sorta duplicative as long as the RN SDK is importing push-fcm.